### ISSUE TITLE
[Feature]: Incorrect and missing Java Addition features

### ISSUE THREAD

**Issue by makarichevss** [mallowigi/material-theme-issues#474]
### Describe the feature

Please make some fixes into Java Additions section in IntelliJ IDEA.
Awesome work is done in 9th version of MT, but something is yet to do.
Please check the numbers on the screenshot and the comments to it below:

1) Please split this into **abstract** and **extends, implements** and **native**. These keywords are from different meaning groups.
2) Please split this into **class** and **interface**. These keywords are from different meaning groups.
3) These are not Java, but Javascript keywords - please delete it from Java Additions.
4) Please split this into **static** and **final**. These keywords are from different meaning groups.
4) Please split this into **this** and **super**. These keywords are from different meaning groups.
6) Now **void** is marked as a primitive type, though it's not. Please make a separate section for **void** and remove it from **Primitive types**.

And some open discussion topics:

- You can delete **assert** keyword - a mark of code smell, it's almost never used in modern Java applications
- **native** is used only in Java documentation, maybe also the candidate for deletion
- **Primitives** section - it's a good idea to separate **null** from **true** and **false** (like in Javascript Primitives section)

![2024-03-31_185056](https://github.com/mallowigi/material-theme-issues/assets/62963044/581aba6e-cc36-4ada-a4ce-88cc66352b86)

### Material Theme Version

v9.1.0

### IDE

IntelliJ IDEA

### System Info

```shell
IntelliJ IDEA 2024.1 RC (Ultimate Edition)
Build #IU-241.14494.158, built on March 21, 2024
You have a perpetual fallback license for this version.
Runtime version: 17.0.10+8-b1207.12 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 11.0
GC: G1 Young Generation, G1 Old Generation
Memory: 4096M
Cores: 24
Registry:
  ide.balloon.shadow.size=0
  editor.distraction.free.mode=true
  editor.focus.mode.color.light=98b6ff
  editor.focus.mode.color.dark=98b6ff
  run.processes.with.pty=false
  suggest.all.run.configurations.from.context=true
  ide.new.editor.tabs.vertical.borders=true
  ide.experimental.ui=true
  terminal.new.ui=true
Non-Bundled Plugins:
  some.awesome (1.15)
  com.obroom.plugin.toolset (2024.3.1)
  com.github.catppuccin.jetbrains (3.2.3)
  com.intellij.grazie.pro (0.3.294)
  google-java-format (1.21.0.0)
  com.intellij.spring.websocket (241.14494.158)
  String Manipulation (9.12.0)
  GrepConsole (12.25.0-IJ2023.3)
  ru.adelf.idea.dotenv (2024.1)
  MavenRunHelper (4.27.1-IJ2022.2)
  com.intellij.ml.llm (241.14494.240)
  izhangzhihao.rainbow.brackets (2024.2.2-241)
  com.intellij.bigdatatools.core (241.14494.240)
  com.haulmont.jpab (241.14494.158)
  com.chrisrm.idea.MaterialThemeUI (9.1.0)
  com.mallowigi (95.0.0)
  mobi.hsz.idea.gitignore (4.5.2)
  com.intellij.bigdatatools.kafka (241.14494.158)
  org.jetbrains.plugins.docker.gateway (241.14494.189)
Kotlin: 241.14494.158-IJ
```

**Comment by mallowigi** [comment]
Thanks for the feedback

**Comment by makarichevss** [comment]
And one more bug:
**class, interface** should mark only class and interface, not **extends** or **implements**

![2024-03-31_212114](https://github.com/mallowigi/material-theme-issues/assets/62963044/784b459a-cffb-445f-b791-d22bb9b8c29e)

**Comment by mallowigi** [comment]
Now that I have some time replying back, I'll currently decline implementing these, for the following reason:

These additions are made in order to give more customization options to the supported languages, however I'm trying not to give too much customization by offering per-keyword customization in order to not complicate too much the options provided.

Therefore these keywords are grouped according to a specific "domain" as much as I can:
- anything related to inheritance (`abstract, implements, extends` - `native` is just an extension for C/JNI based applications if I remember correctly)
- `assert` for basic assertions. I know this is a frowned upon pattern but whatever.
- simple types, including `void`
- OOP: `class` and `interface`
- `enum` has its own category, but logically it should have been put with OOP I guess
- `for`, `while` and `do` are related to loops
- `if`, `else` conditional
- `import` and `package` for package management
- `instanceof` for type checking
- `module`, `requires` etc... these are not just JavaScript keywords, but Java as well, specifically for Java 9 modules feature
- `new` for object creation
- `null`, `true` and `false` for primitive values. I could further separate `null` from `true` and `false` as you stated.
- `private`, `public`, `protected` for modifiers
- `return`
- `static` and `final`. I admit that these are not related. I'll try to provide different categories
- `switch`, `case`, `default` for pattern matching
- `synchronized`, `volatile`, `transient` - related to threading
- `this` and `super`: purely by choice, as these are not keywords but more like "instances" - `this` being the current object and `super` being the parent object. They are in the same logical group and I don't plan to separate them. It's also the case for all languages.
- `throw`, `throws` - related to exceptions
- `try`, `catch`, `finally` - obvious
- `var` for dynamic typing

So to sum it up, the only changes I'll make will be to separate `null` from `true, false` and `static` from `final`.
